src/libraries/System.Memory/tests/ReadOnlySpan/Split.T.cs

```diff
@@ -0,0 +1,220 @@
+            [ (CustomStruct[])[new(1), new(2), new(3)], default(CustomStruct), (Range[])[0..3] ],
+            [ (CustomClass[])[new(1), new(2), new(3)],  default(CustomClass),  (Range[])[0..3] ],
+
+            // Split no matching element
+            [ (char[])['a', ' ', 'b'], ',', (Range[])[0..3] ],
+            [ (int[]) [1, 2, 3], (int)4,    (Range[])[0..3] ],
+            [ (long[])[1, 2, 3], (long)4,   (Range[])[0..3] ],
+            [ (byte[])[1, 2, 3], (byte)4,   (Range[])[0..3] ],
+            [ (CustomStruct[])[new(1), new(2), new(3)], new CustomStruct(4), (Range[])[0..3] ],
+            [ (CustomClass[])[new(1), new(2), new(3)],  new CustomClass(4),  (Range[])[0..3] ],
+
+            // Split on sequence containing only a separator
+            [ (char[])[','], ',',     (Range[])[0..0, 1..1] ],
+            [ (int[]) [1], (int)1,    (Range[])[0..0, 1..1] ],
+            [ (long[])[1], (long)1,   (Range[])[0..0, 1..1] ],
+            [ (byte[])[1], (byte)1,   (Range[])[0..0, 1..1] ],
+            [ (CustomStruct[])[new(1)], new CustomStruct(1), (Range[])[0..0, 1..1] ],
+            [ (CustomClass[]) [new(1)], new CustomClass(1),  (Range[])[0..0, 1..1] ],
+
+            // Split on empty sequence with default separator
+            [ (char[])[], default(char), (Range[])[0..0] ],
+            [ (int[]) [], default(int),  (Range[])[0..0] ],
+            [ (long[])[], default(long), (Range[])[0..0] ],
+            [ (byte[])[], default(byte), (Range[])[0..0] ],
+            [ (CustomStruct[])[], default(CustomStruct), (Range[])[0..0] ],
+            [ (CustomClass[]) [], default(CustomClass),  (Range[])[0..0] ],
+
+            [ (char[])['a', ',', 'b'], ',', (Range[]) [ 0..1, 2..3 ] ],
+            [ (int[]) [1, 2, 3], (int)2,    (Range[]) [ 0..1, 2..3 ] ],
+            [ (long[])[1, 2, 3], (long)2,   (Range[]) [ 0..1, 2..3 ] ],
+            [ (byte[])[1, 2, 3], (byte)2,   (Range[]) [ 0..1, 2..3 ] ],
+            [ (CustomStruct[])[new(1), new(2), new(3)], new CustomStruct(2), (Range[]) [ 0..1, 2..3 ] ],
+            [ (CustomClass[])[new(1), new(2), new(3)],  new CustomClass(2),  (Range[]) [ 0..1, 2..3 ] ],
+
+            [ (char[])['a', 'b', ',', ','], ',', (Range[]) [ 0..2, 3..3, 4..4 ] ],
+            [ (int[]) [1, 3, 2, 2], (int)2,      (Range[]) [ 0..2, 3..3, 4..4 ] ],
+            [ (long[])[1, 3, 2, 2], (long)2,     (Range[]) [ 0..2, 3..3, 4..4 ] ],
+            [ (byte[])[1, 3, 2, 2], (byte)2,     (Range[]) [ 0..2, 3..3, 4..4 ] ],
+            [ (CustomStruct[])[new(1), new(3), new(2), new(2)], new CustomStruct(2), (Range[]) [ 0..2, 3..3, 4..4 ] ],
+            [ (CustomClass[])[new(1), new(3), new(2), new(2)],  new CustomClass(2),  (Range[]) [ 0..2, 3..3, 4..4 ] ],
+        ];
+
+        [Theory]
+        [MemberData(nameof(SplitSingleElementSeparatorData))]
+        public static void Split_SingleElementSeparator<T>(T[] value, T separator, Range[] result) where T : IEquatable<T>
```

Comment:
I believe we're going to need to disable these tests on Native AOT due to the use of generics in a way it can't see.

Reply:
Aw shoot, I hadn't considered that. I presume that is the reason most other tests in ReadOnlySpanTests don't use generics?

Reply:
Alternatively I can unroll the tests into their own non generic files.

Reply:
It just needs a small bit of refactoring to call the various logic explicitly rather than via reflection. I pushed a commit to do so.